pkg/cmd/codespace/edit.go

```diff
@@ -44,6 +44,27 @@ func (a *App) Edit(ctx context.Context, opts editOptions) error {
 		DisplayName:   opts.displayName,
 		SKU:           opts.machine,
 	}
+
+	if userInputs.DisplayName == "" && userInputs.SKU == "" {
+		return fmt.Errorf("please pass in at least one valid argument to be edited")
+	}
+
+	if userInputs.CodespaceName == "" {
+		a.StartProgressIndicatorWithLabel("Fetching codespaces")
+		codespaces, err := a.apiClient.ListCodespaces(ctx, -1)
+		a.StopProgressIndicator()
+		if err != nil {
+			return fmt.Errorf("error getting codespaces to select from: %w", err)
+		}
+
+		selectedCodespace, err := chooseCodespaceFromList(ctx, codespaces)
+		if err != nil {
+			return fmt.Errorf("error choosing codespace: %w", err)
+		}
```

Comment:
There is a higher-level method that combines the API call and the prompt, see: https://github.com/cli/cli/blob/e837b7bc4d5717f7df31f2e77b813f6e260fadbb/pkg/cmd/codespace/common.go#L115